haystack/nodes/reader/base.py

```diff
@@ -105,10 +105,12 @@ def run(self, query: str, documents: List[Document], top_k: Optional[int] = None
 
     def run_batch(  # type: ignore
         self,
-        queries: List[str],
+        queries: Union[str, List[str]],
         documents: Union[List[Document], List[List[Document]]],
         top_k: Optional[int] = None,
         batch_size: Optional[int] = None,
+        labels: Optional[List[MultiLabel]] = None,
+        add_isolated_node_eval: bool = False,
```

Comment:
Do we need this parameter `add_isolated_node_eval`? As a user of this API it wasn't clear to me immediately what it is about and why do we need it?

Reply:
Yes, we need it. It's the same parameter as in the standard `run()`. If it is set to `True`, the evaluation is executed with labels as node inputs in addition to the integrated evaluation, where the node inputs are the outputs of the previous node in the pipeline.